model/src/test/kotlin/config/OrtConfigurationTest.kt

```diff
@@ -85,5 +87,34 @@ class OrtConfigurationTest : WordSpec({
                 scanner.options shouldNotBe null
             }
         }
+
+        "correctly prioritize the sources" {
+            val configFile = createTempFile(ORT_NAME, javaClass.simpleName)
+            configFile.deleteOnExit()
+            configFile.writeText(
+                """
+                ort {
+                  scanner {
+                    postgresStorage {
+                      url = "postgresql://your-postgresql-server:5444/your-database"
+                      schema = schema
+                      username = username
+                      password = password
+                    }
+                  }
+                }
+                """.trimIndent()
+            )
+
+            val config = OrtConfiguration.load(
+                configFile,
+                mapOf("ort.scanner.postgresStorage.schema" to "argsSchema")
+            )
+
+            config.scanner shouldNotBe null
+            config.scanner!!.postgresStorage shouldNotBe null
+            config.scanner!!.postgresStorage!!.username shouldBe "username"
```

Comment:
Is `scanner!!` still needed here as we already have it above? Similar below.

Reply:
Yes, otherwise you get: `Smart cast to 'ScannerConfiguration' is impossible, because 'config.scanner' is a public API property declared in different module`